client/repo.c

```diff
@@ -555,6 +555,48 @@ TDNFGetRepoMD(
                   pszRepoDataDir);
     BAIL_ON_TDNF_ERROR(dwError);
 
+    if (pRepoData->pszMirrorList) {
+        time_t now = time(NULL);
+        int needDownload = 0;
+        struct stat st = {0};
+
+        dwError = TDNFGetCachePath(pTdnf, pRepoData,
+                                   TDNF_REPO_METADATA_MIRRORLIST, NULL,
+                                   &pszMirrorFile);
+        BAIL_ON_TDNF_ERROR(dwError);
+
+        if (stat(pszMirrorFile, &st) < 0) {
+            if (errno == ENOENT)
+                needDownload = 1;
+            else {
+                dwError = errno;
+                BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
+            }
+        } else if ((now - st.st_ctime) > pRepoData->lMetadataExpire)
+            needDownload = 1;
+
+        if (needDownload) {
+            dwError = TDNFDownloadFile(pTdnf, pRepoData, pRepoData->pszMirrorList, pszMirrorFile, pRepoData->pszId);
+            BAIL_ON_TDNF_ERROR(dwError);
+        }
+
+        dwError = TDNFReadFileToStringArray(pszMirrorFile, &pRepoData->ppszBaseUrls);
+        BAIL_ON_TDNF_ERROR(dwError);
+
+        /* remove comments from mirror file */
+        int i, j;
+        for(i = 0, j = 0; pRepoData->ppszBaseUrls[i]; i++) {
+            if (pRepoData->ppszBaseUrls[i][0] == '#')
```

Comment:
use strspn here and ignore empty lines, `#` preceded by spaces or tabs
or is it omitted at the time of populating this buffer?

Reply:
I need to check if that is allowed in mirror files.

Reply:
I thought about this, I think I leave it as is. I wasn't able to find out how dnf parses the mirror file, but I will asume that a comment has to start with `#` at the beginning of a line.